src/vsr/replica.zig

```diff
@@ -4993,13 +4896,27 @@ pub fn ReplicaType(
             // The view/log_view incremented while the previous view-change update was being saved.
             const update = self.log_view_durable() < self.log_view or
                 self.view_durable() < self.view;
-
             const update_dvc = update and self.log_view < self.view;
             const update_sv = update and self.log_view == self.view and
                 (self.replica != self.primary_index(self.view) or self.status == .normal);
             assert(!(update_dvc and update_sv));
 
             if (update_dvc or update_sv) self.view_durable_update();
+
+            // Reset SVC timeout in case the view-durable update took a long time.
+            if (self.view_change_status_timeout.ticking) self.view_change_status_timeout.reset();
+
+            // Trigger work that was deferred until after the view-change update.
+            if (self.status == .normal) {
+                if (self.primary_index(self.view) == self.replica) {
+                    const start_view = self.create_view_change_message(.start_view);
+                    defer self.message_bus.unref(start_view);
+
+                    self.send_message_to_other_replicas(start_view);
+                } else {
+                    self.send_prepare_oks_after_view_change();
+                }
+            }
```

Comment:
Shouldn't we also `self.send_do_view_change()` here `if self.start_view_change_quorum` ?